testing prince with default simulated data

dieloz opened the pull request:
I was about to test PRINCE with default simulated data but I encounter a couple of issues. Could you please take a look and let me know whether all looks fine?
Best,
Diego

dieloz committed:
testing prince with default simulated data

rubvber commented:
Hi Diego,

Thanks for reaching out. I have (manually) merged your deletion of p.singletau on line 312 into the code as this was indeed a mistake (p.singletau is a legacy parameter from an earlier version). As for your other two proposed changes: 

1. Changing the default value of p.dec_type to 'PRINCE' is something you could do in your own code, but in general I would prefer to leave the default to 'TAFKAP' (the newer version) and allow users to specify their preference through the parameter structure 'p' (so e.g. p = struct('dec_type', 'PRINCE'); TAFKAP_Decode(data, p);).

2. Omitting the selection of the first p.nchan columns of W on line 238 can cause errors when p.prev_C=true. In that case, the final p.nchan columns of W reflect the basis function coefficients for a potential lingering response to the previous stimulus, and we only want to select the first p.nchan columns for use in the covariance model. So while this selection may seem useless in your own use case, it is important for the code to work correctly in general. 

Note that the code should work correctly without those two changes - if it doesn't please let me know what error(s) you are getting.

Best,
Ruben

dieloz committed:
minor fix: reply to Ruben + shuffle_oris seems to be irrelevant

dieloz committed:
Merge branch 'jeheelab:main' into test_prince

dieloz commented:
Hi Ruben,
Thanks for your response. Let me reply to your points
1. I fully agree with you. Indeed, setting 'PRINCE' as the default was my mistake: I didn't realize before submitting the pull request
2. You're fully right!
With these changes you proposed it's all right. In the new commit, I added them and I removed the "shuffle_oris" which it seems legacy
Best
